libraries/provider_plugin.rb

```diff
@@ -0,0 +1,47 @@
+class Chef
+  class Provider::ElasticsearchPlugin < Provider
+    include Poise
+    include ElasticsearchCookbook::Helpers
+    include Chef::Mixin::ShellOut
+
+    def action_install
+      name    = new_resource.plugin_name
+      version = new_resource.version ? "/#{new_resource.version}" : nil
+      url     = new_resource.url     ? " -url #{new_resource.url}" : nil
+
+      # nasty workaround to find a nice default for plugin dir
+      unless new_resource.plugin_dir
+        found_installs = run_context.resource_collection.select { |r| r.instance_variable_get(:@resource_name) == :elasticsearch_install }
+
+        unless found_installs.empty?
+
+          install_resource = found_installs.first
+          es_version = install_resource.instance_variable_get(:@version)
+          es_dir = get_source_root_dir(install_resource, node)
+
+          new_resource.plugin_dir "#{es_dir}/elasticsearch-#{es_version}/plugins"
+        end
+      end
+
+      raise 'Could not determine the plugin directory. Please set plugin_dir on this resource.' unless new_resource.plugin_dir
+
+      return if Dir.entries(new_resource.plugin_dir).any? do |plugin|
+        next if plugin =~ /^\./
+        name.include? plugin
+      end rescue false
```

Comment:
I'm assuming it is to make the chef code return/exit, vs. raise an exception. There's no nice (idempotent) way to ask `bin/plugin` to tell us if a plugin is already installed is there? It would be super if there was some way to say, "Install if we don't have it," but as far as I know, the plugin install binary that comes with ES doesn't do this.